### DISCUSSION TITLE
--workdir 参数的是相对于项目目录而不是当前目录

### DISCUSSION THREAD

**Discussion by q962** [xmake-io/xmake#5857] [General]
如果在子目录中运行时指定 `.` 就会得到不符合直觉的情况。

**Comment by waruqi** [comment]
只要是 xmake 命令，不管在任何子目录执行，都是以项目根目录作为 workdir 。。不是是否符合直觉问题，目前的设计，压根不推荐在子目录执行。

**Reply by q962** [reply]
你想表达的是：脚本的工作目录是项目根目录？

或许我的描述不够明确。

我不清楚是否还有其他命令也提供 --workdir 参数。

我想说的是 `xmake r -w .` 在子目录下执行时 --workdir 参数的是相对于项目目录而不是当前目录。
